src/index.ts

```diff
@@ -153,29 +153,67 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
 
   if (response) {
     let headers = new Headers(response.headers)
-    headers.set('CF-Cache-Status', 'HIT')
     if (shouldSetBrowserCache) {
       headers.set('cache-control', `max-age=${options.cacheControl.browserTTL}`)
     } else {
       // don't assume we want same cache behavior of edge TTL on client
       // so remove the header from the response we'll return
       headers.delete('cache-control')
     }
-    response = new Response(response.body, { headers })
+
+    let shouldRevalidate = false
+    // Four preconditions must be met for a 304 Not Modified:
+    // - the request cannot be a range request
+    // - client sends if-none-match
+    // - resource has etag
+    // - test if-none-match against the pathKey so that we test against KV, rather than against
+    // CF cache, which may modify the etag with a weak validator (e.g. W/"...")
+    shouldRevalidate = [
+      request.headers.has('range') !== true,
+      request.headers.has('if-none-match'),
+      response.headers.has('etag'),
+      request.headers.get('if-none-match') === `${pathKey}`,
+    ].every((val) => val === true)
```

Comment:
Hey @SukkaW - thanks for this too.

I suppose I could be persuaded otherwise, but `.every()` uses the strict equality algorithm and allows for short circuiting. Your suggestion is equivalent to the existing solution, but it is not faster

Reply:
@shagamemnon I had run some benchmark. The suggestion I given is about 8% faster. Not much though.

Also, `&&` support shorthand algorithm as well.